src/libs/TransactionUtils/index.ts

```diff
@@ -814,41 +814,37 @@ function isOnHoldByTransactionID(transactionID: string): boolean {
 /**
  * Checks if any violations for the provided transaction are of type 'violation'
  */
-function hasViolation(transactionID: string, transactionViolations: OnyxCollection<TransactionViolations>): boolean {
+function hasViolation(transactionID: string, transactionViolations: OnyxCollection<TransactionViolations>, showInReview?: boolean): boolean {
     return !!transactionViolations?.[ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS + transactionID]?.some(
-        (violation: TransactionViolation) => violation.type === CONST.VIOLATION_TYPES.VIOLATION,
+        (violation: TransactionViolation) => violation.type === CONST.VIOLATION_TYPES.VIOLATION && (showInReview === undefined || showInReview === (violation.showInReview ?? false)),
```

Comment:
This logic is confusing. If `showInReview` is `false` and `violation.showInReview` is unset (not sent by BE). Then we will end up showing this in review? (false === false)

Reply:
Ah I see it now, that's a filter. 

<img src="https://media.tenor.com/_rrC613KIJMAAAAM/the-simpsons-homer-simpson.gif" />